### ISSUE TITLE
Refactor `core.#Stop` and `core.#SendSignal` to not reference `core.#Start` directly

### ISSUE THREAD

**Issue by helderco** [dagger/dagger#2776]
## Problem

We want to protect the user from trying to nest core actions since they'll be ignored for execution. As described in the following issue, the `#Stop` and `#SendSignal` core actions reference `#Start` as a valid use case here:

- https://github.com/dagger/dagger/issues/2303#issuecomment-1176529119

## Solution

Refactor out this pattern to avoid referencing another core action directly. Most likely using a field from it instead.

**TODO:** design proposal

/cc @sipsma

**Comment by marcosnils** [comment]
How did this work initially since `SendSignal` and `Stop` always referenced `Start` since it's early design?

**Comment by helderco** [comment]
Sorry, I don't understand the question. Nothing has been changed yet.

**Comment by marcosnils** [comment]
> Sorry, I don't understand the question. Nothing has been changed yet.

IIUC this issue is about changing how `#Stop` and `#SendSignal` work so we don't have to reference the `#Start` action since it triggers the ignore issue correct? My question/observation is that since its early design, that's the way that `#Stop` and `#SendSignal` worked, by referencing the `#Start` action. How come we didn't notice that this was an issue before coming up with this UX if we knew about the ignore issue beforehand? 

I think there's something that I'm not quite understanding here.

**Comment by helderco** [comment]
> How come we didn't notice that this was an issue before coming up with this UX if we knew about the ignore issue beforehand?

TLDR: This feature was created before the validation issue was flagged, not the other way around. Even then, I don't think anyone connected the two.

IIRC @sipsma needed explicit dependencies but that hasn't been done yet. It works between core actions, but not composite actions so we never advertised this but know it internally[^1]. He learned about this and used it to make progress on this feature. Notice that he submitted his PR in **April 16th**:

- https://github.com/dagger/dagger/pull/2214#issuecomment-1100479660

Afterwards, in **April 25th**, @aluzzardi submitted an issue because some people were defining their core actions' dependencies inside them, so it followed that we should validate against this unsupported use case. 

I'm not sure if anyone connected both issues (validation for trying to **execute nested core actions** vs using a core action **reference** as a way to create an explicit dependency for **execution order** in our code base), until I tried to validate this in July 6th:

> A nested core action can be used to force a dependency on another core action. **Execution is ignored as expected**, but the other core action is recognized as a dependency and therefore runs first.

- https://github.com/dagger/dagger/issues/2303#issuecomment-1176529119

[^1]: Notice that I've used it in `client: filesystem` as well, but that was easy to fix because it's hidden, not user facing.

**Comment by helderco** [comment]
Closing since the 0.2 CUE API is frozen.
